backend/src/main/java/ai/verta/modeldb/versioning/VersioningUtils.java

```diff
@@ -220,7 +221,7 @@ public static String getVersioningCompositeId(
     }
   }
 
-  public static void saveOrUpdateArtifactPartEntity(
+  public static void saveArtifactPartEntity(
       ArtifactPart artifactPart, Session session, String artifactId, int artifactType) {
     ArtifactPartEntity artifactPartEntity =
         new ArtifactPartEntity(
```

Comment:
@conradoverta the bug fi is in this method.  There was originally a call to 
```
session.lock(artifactEntityPart, LockMode.PESSIMISTIC_WRITE);
```
But since the object is always new this would fail.  Removing the attempt to lock the object succeeds.